mysql-test/lib/mtr_report.pm

```diff
@@ -497,29 +497,40 @@ sub mtr_report_stats ($$$$) {
       $test_time = sprintf("%.3f", $test->{timer} / 1000);
       $test->{'name'} =~ s/$current_suite\.//;
 
-      my $test_result;
+      my $test_result= $test->{'result'};
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
       if ($test->{'retries'} > 0) {
-        $test_result = "MTR_RES_FAILED";
+        if ($test->{'result'} eq "MTR_RES_PASSED"){
+          $test_result = "MTR_RES_RETRY_PASSED";
+        }
+        else {
+          if ($test->{'failures'} > 1){
+            $test_result = "MTR_RES_RETRY_FAILED"; # retry failure
+          }
```

Comment:
what would be the status here? if `$test->{result}` is not `"MTR_RES_PASSED"` and `$test->{failures} <= 1` ?

Reply:
Test result is set at the beginning `my $test_result= $test->{'result'};`.
In situation where `$test->{failures} = 1` it will be `MTR_RES_FAILED` .

Reply:
Yes, and is it correct to have the status `MTR_RES_FAILED` when `$test->{'retries'} > 0`, that is, when it's clearly a retry?

Reply:
IMO it is since stdout is generating the same when `$test->{'retries'} > 0 ` and the failure is first one.
Proof: https://github.com/MariaDB/server/pull/1753#issuecomment-808366027

Reply:
Try it. As far as I understand on the first failure `$test->{retries}` is not set at all.

Reply:
It wasn't but this patch changed that, since we saved the result, we saved the retries to be a default (2) before invoking `mtr_report`: https://github.com/MariaDB/server/pull/1753/files#diff-3b008cd0a0b878ff56de8f6187554a266ecca2140c4e4fc56758dfa1dc2eba6cR932